avocado/utils/lv_utils.py

```diff
@@ -488,14 +488,14 @@ def lv_revert(vg_name, lv_name, lv_snapshot_name):
         # and attempt to reactivate the volume.
         active_lv_pattern = re.escape("%s [active]" % lv_snapshot_name)
         lvdisplay_output = process.system_output("lvdisplay", sudo=True)
-        if ('Snapshot could not be found' in ex and
+        if ('Snapshot could not be found' in ex.result.stderr and
                 re.search(active_lv_pattern, lvdisplay_output) or
-                "The Logical volume %s is still active" % lv_name in ex):
+                "The Logical volume %s is still active" % lv_name in ex.result.stderr):
             LOGGER.debug(("Logical volume %s is still active! " +
                           "Attempting to deactivate..."), lv_name)
             lv_reactivate(vg_name, lv_name)
             LOGGER.error("Continuing after reactivation")
-        elif 'Snapshot could not be found' in ex:
+        elif 'Snapshot could not be found' in ex.result.stderr:
             LOGGER.error(ex)
```

Comment:
this should be `LOGGER.error(ex.result)` ?

Reply:
Not sure what's the perfect message to be logged here, but indeed `ex.result` seems like a better option.  I'm changing the order of lines 498 and 499, though, to make a "heading" precede the command output.  I mean:

```diff
diff --git a/avocado/utils/lv_utils.py b/avocado/utils/lv_utils.py
index 016cc187..ff2806db 100644
--- a/avocado/utils/lv_utils.py
+++ b/avocado/utils/lv_utils.py
@@ -496,8 +496,8 @@ def lv_revert(vg_name, lv_name, lv_snapshot_name):
             lv_reactivate(vg_name, lv_name)
             LOGGER.error("Continuing after reactivation")
         elif 'Snapshot could not be found' in ex.result.stderr:
-            LOGGER.error(ex)
-            LOGGER.error("Could not revert to snapshot")
+            LOGGER.error("Could not revert to snapshot:")
+            LOGGER.error(ex.result)
         else:
             raise ex
 
```